app/src/main/java/com/example/saar/Gallery/GalleryFragment.java

```diff
@@ -32,6 +34,14 @@
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
+        if (!isNetworkConnected()) {
+            Fragment fragment = new NoInternetFragment();
+            if (fragment != null && getActivity() != null) {
```

Comment:
@somenath1435 why don't u simply use an image to replace it. Why r u using a different fragment. It would be better if u keep an image view initially Visibility GONE. If no internet connection is found, then simply make visibility visible. It would prevent memory loss.